Assets/Scripts/UI/GameUI.cs

```diff
@@ -3,35 +3,101 @@
 using UnityEngine;
 using UnityEngine.UI;
 using Common;
+using UnityEngine.SceneManagement;
+using Inputs;
+
+public enum GameState { MainMenu, Game, GameOver }
 
 namespace UI
 {
     public class GameUI : Singleton<GameUI>
     {
-        [SerializeField] private Text playerScoreText, playerHealthText;
+        [SerializeField] private DefaultScriptableObject defaultScriptableObject;
+        [SerializeField] private Text playerScoreText, playerHealthText, hiScoreText, gameOverScoreText;
+        [SerializeField] private GameObject gameMenu, mainMenu, gameOverMenu;
+
+        private GameState gameState = GameState.MainMenu;
+
+        public GameState GameState { get { return gameState; }}
 
         private int playerScore = 0;
 
+        void OnEnable()
+        {
+            SceneManager.sceneLoaded += OnSceneLoaded;
+        }
+
+        void OnDisable()
+        {
+            SceneManager.sceneLoaded -= OnSceneLoaded;
+        }
+
+        private void OnSceneLoaded(Scene scene, LoadSceneMode mode)
+        {
+            switch (gameState)
+            {
+                case GameState.MainMenu:
+                    mainMenu.SetActive(true);
+                    hiScoreText.text = "HiScore:" + defaultScriptableObject.highScore;
+                    break;
+                case GameState.Game:
+                    mainMenu.SetActive(false);
+                    gameMenu.SetActive(true);
+                    gameOverMenu.SetActive(false);
+                    break;
+                case GameState.GameOver:
+                    gameMenu.SetActive(false);
+                    gameOverMenu.SetActive(true);
+                    gameOverScoreText.text = "PlayerScore:" + playerScore;
+                    InputManager.Instance.EmptyInputComponentList();
+                    StartCoroutine(Respawn());
```

Comment:
lets change this to let user control if they want to go to the lobby or back into the game

Reply:
Done.